src/path.c

```diff
@@ -1887,7 +1887,7 @@ extern int git_path_is_gitfile(const char *path, size_t pathlen, git_path_gitfil
 	const char *file, *hash;
 	size_t filelen;
 
-	if (gitfile < 0 && gitfile >= ARRAY_SIZE(gitfiles)) {
+	if (!(gitfile > 0 && gitfile <= ARRAY_SIZE(gitfiles))) {
```

Comment:
But wait, this is wrong. It's now equivalent to `gitfile <= 0 || gitfile > ARRAY_SIZE(gitfiles)`. So if `gitfile == 0`, we fail even though it's a valid enum value